File: Data Structures & Algorithms/remove-element/submission-2.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        left_ptr = 0
        right_ptr = len(nums) - 1

        while left_ptr <= right_ptr:
            if nums[left_ptr] == val:
                temp = nums[left_ptr]
                nums[left_ptr] = nums[right_ptr]
                nums[right_ptr] = temp
                right_ptr -= 1
            else:
                left_ptr += 1
        return left_ptr

File: Data Structures & Algorithms/concatenation-of-array/submission-2.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> ans(nums.size() * 2);
        int midIndex = nums.size();

        for (int i = 0; i < midIndex; i++) {
            ans[i] = nums[i];
            ans[i + midIndex] = nums[i];
        }
        return ans;
    }
};